### PR TITLE
30 improve markable position bb

### PR DESCRIPTION
**Pull request by barrybecker4**
Made suggested changes and added unit tests.
Also replaced HashSet with Set in the interface.

### CODE REVIEW

**Comment by fsparv** [comment]
conceptually this looks good. I'll need to pull it down, run the test with old/new code etc. I think I might be able to get to this and other pull requests later today (time for sleep now... or rather several hours ago :) ).

File: src/test/java/leelawatcher/goboard/MarkablePositionTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        assertFalse(pos.isMarked(p1));
+    }
+
+    @Test
+    public void testClearMarks() {
+        MarkablePosition pos = new MarkablePosition(new Position());
+        PointOfPlay p1 = new PointOfPlay(2, 2);
+        PointOfPlay p2 = new PointOfPlay(2, 3);
+        pos.setMark(p1);
+        pos.setMark(p1);
```
Comment (fsparv) [inline src/test/java/leelawatcher/goboard/MarkablePositionTest.java]: why set p1 twice?

File: src/test/java/leelawatcher/goboard/MarkablePositionTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        pos.setMark(new PointOfPlay(2, 2));
+        assertTrue(pos.isMarked(new PointOfPlay(2, 2)));
+    }
+
+    @Test
+    public void testClearMark() {
+        MarkablePosition pos = new MarkablePosition(new Position());
+        PointOfPlay p1 = new PointOfPlay(2, 3);
+        pos.setMark(p1);
+        pos.setMark(p1);
```
Comment (fsparv) [inline src/test/java/leelawatcher/goboard/MarkablePositionTest.java]: why set p1 twice here?

**Review by fsparv — CHANGES_REQUESTED** [review]
Seems good overall. If you make a couple minor tweaks noted below I'm happy to merge it.

File: src/main/java/leelawatcher/goboard/MarkablePosition.java [hunk 3]
```diff
@@ -153,67 +146,33 @@ private boolean isOnBoard(PointOfPlay p, int boardSize) {
...
+        checkNbr(p, p.getX() + 1, p.getY(), members, boardSize);
+        checkNbr(p, p.getX(), p.getY() - 1, members, boardSize);
+        checkNbr(p, p.getX() - 1, p.getY(), members, boardSize);
       }
     }
     return members;
   }
 
-}
+  private void checkNbr(PointOfPlay p, int x, int y, Set<PointOfPlay> members, int boardSize) {
```
Comment (fsparv) [inline src/main/java/leelawatcher/goboard/MarkablePosition.java]: Please write out whatever Nbr stands for. I like long method names, my fingers will survive the typing but my brain might not survive the confusion... :)

**Comment by barrybecker4** [comment]
OK. Made the requested changes.

**Pull request merged**